tests/test_methods.rs

```diff
@@ -231,11 +231,15 @@ impl MethArgs {
         [a.to_object(py), args.into(), kwargs.to_object(py)].to_object(py)
     }
 
+    #[args("*", c = 10)]
```

Comment:
Should you also add a test for `args(a, "*", c=10)`? (or whatever the correct syntax is for a required argument, I'm not sure).

Reply:
Thanks.
Now this allows `args(a, "*", c=10)`, `args(a, b, "*", c=10)`, and `args("*", c=10)`, which seems confusing :thinking: 
So it may be good to restrict the syntax :thinking:

Reply:
I don't think it's a good idea to restrict things, and in fact we should also add support for positional-only arguments (in a separate change, IMO). The full spec for a python function definition can be found here: https://docs.python.org/3/tutorial/controlflow.html#special-parameters

```
def f(pos1, pos2, /, pos_or_kwd, *, kwd1, kwd2):
      -----------    ----------     ----------
        |             |                  |
        |        Positional or keyword   |
        |                                - Keyword only
         -- Positional only
```

That doesn't include `*args` and `**kwargs`.

Reply:
Yeah, but our proc-macro has a somewhat more free syntax like
```rust
    #[args(a, "*", d=10)]
    fn get_pos_arg_kw_sep(&self, a: i32, b: i32, c: i32) -> PyResult<i32>
```
, which is really confusing.

Reply:
If we're working on this, should we also change to accept `*` instead of `"*"` ? e.g. `#[args(a, *, d=10)]`

Reply:
(Maybe we should note these down each as separate bug issues.)

Reply:
syn doesn't accept it.

Reply:
I could be wrong, can't it parse it as `syn::Token![*]` ?

Reply:
https://docs.rs/syn/1.0.16/syn/enum.NestedMeta.html